### ISSUE TITLE
Extend UserManager<T> class for Custom Password Hashing service

### ISSUE THREAD

**Issue by kassim-shaikh** [aspnet/Identity#1842]
Hi,
I am using Identity framework for user profile management in my MVC application. There are certain aspects  of the identity framework which  i want to customize based on application need. One of them is **Custom password hashing** :

I am using combination of Salt, DEK and MEK to generate the random salt and encrypt it with random DEK once the hashing is done. I can't find a way to incorporate this feature just by extending the `IPasswordHashing` interface. One way i could achieve is by extended the `UserManager<T>` class. 

Could any one help me with the best practice to extend the class or basically where can i put custom hashing code.

**Comment by blowdart** [comment]
https://github.com/aspnet/Identity/blob/dev/src/Core/IPasswordHasher.cs is the interface for hashers.

Then shove it in DI after you've added Identity,

services.AddSingleton<IPasswordHasher<TApplication>, YourPasswordHasher<TApplication>>();

**Comment by blowdart** [comment]
We're closing this issue as no response or updates have been provided in a timely manner. If you have more details and are encountering this issue please add a new reply and re-open the issue.

**Comment by aspnet-hello** [comment]
We periodically close 'discussion' issues that have not been updated in a long period of time.

We apologize if this causes any inconvenience. We ask that if you are still encountering an issue, please log a new issue with updated information and we will investigate.
